packages/ui/src/TextField.tsx

```diff
@@ -314,6 +314,9 @@ export function TextField({
                     } else {
                       onChange({value: formattedPhoneNumber});
                     }
+                  } else if (type === "number") {
+                    text.replace(/[^0-9]/g, "");
+                    onChange({value: !isNaN(parseInt(text)) ? parseInt(text).toString() : ""});
```

Comment:
This seems to work like I'd expect, but I would have thought it'd be something like ```if (!isNaN(parseInt(text)) { onChange(...) }``` and this code would clear the text completely as soon as you entered a non-number. But it doesn't. Weird.

Reply:
Oh! It's because `parseInt` is smart? 

```
> parseInt("123abc")
123
```

I'm shocked JS is smart about this.

Reply:
haha indeed

Reply:
Based on above, this code can just be `onChange({value: parseInt(text).toString()})` because `isNaN(parseInt())` will never be false.

Reply:
ah yeah true

Reply:
oop, yeah that commit was a bit delayed getting to the system for some reason. done!

Reply:
mm actually i just tested this and parseInt(undefined) or parseInt(null) or parseInt("fdfa") all return NaN. So I added the check back in.